nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -216,10 +224,23 @@ in
 
         requires = lib.optionals cfg.database.createLocally [ "postgresql.service" ];
 
+        path = mkIf (cfg.secretFile != null) [ pkgs.jq ];
+
+        # merge the two configs and prevent others from reading the result
+        preStart = mkIf (cfg.secretFile != null) ''
+          set -u
```

Comment:
Why is `-u` needed here?

Reply:
To make the script fail if `$CREDENTIALS_DIRECTORY` is not set instead of silently expanding to `/secretFile`, I will add a comment.